crates/bevy_ecs/src/world/mod.rs

```diff
@@ -444,6 +444,29 @@ impl World {
     /// assert_eq!(world.get::<Position>(entities[0]).unwrap(), &Position { x: 1.0, y: 0.0 });
     /// assert_eq!(world.get::<Position>(entities[1]).unwrap(), &Position { x: 0.0, y: 1.0 });
     /// ```
+    ///
+    /// To iterate over entities in a deterministic order,
+    /// sort the results of the query using the desired component as a key.
+    ///
+    /// ```
+    /// use bevy_ecs::{entity::Entity, world::World};
+    /// let mut world = World::new();
+    /// let a = world.spawn().insert_bundle((2, "abc")).id();
+    /// let b = world.spawn().insert_bundle((3, "xyz")).id();
+    /// let c = world.spawn().insert_bundle((1, "def")).id();
+    /// let mut entities = world.query::<(Entity, &i32, &str)>()
```

Comment:
I think a `&str` component is not possible

Reply:
Yes, fixing it right now. Noticed that running `cargo test --doc` doesn't run this example, but `cargo test --workspace` does. At the same time the last command point that running tests with `--doc` only will rerun that test.